src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -8657,6 +8657,51 @@ bool MergeTreeData::initializeDiskOnConfigChange(const std::set<String> & new_ad
     return true;
 }
 
+void MergeTreeData::loadPrimaryKeys()
+{
+    // Define the states of parts that need to be processed
+    DataPartStates affordable_states = { MergeTreeDataPartState::Active, MergeTreeDataPartState::Outdated, MergeTreeDataPartState::Deleting };
+
+    // Thread pool to process loading in parallel
```

Comment:
Given that `SYSTEM LOAD INDEX` is supposed to be used for benchmarks, experimentation etc. instead during normal database usage, I'd say we could get rid of concurrent loading here for simplicity and maintainability. It is basically optimizing a rarely used code path.

Reply:
I've updated the loadPrimaryKeys function to remove the thread pool and parallel processing as suggested. Now the primary keys are loaded sequentially, which simplifies the code path.